features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/impl/DatadogSpanBuilderAdapterTest.kt

```diff
@@ -0,0 +1,158 @@
+ * Copyright 2016-Present Datadog, Inc.
+ */
+package com.datadog.android.trace.impl
+
+import com.datadog.android.trace.api.span.DatadogSpan
+import com.datadog.android.trace.api.span.DatadogSpanLink
+import com.datadog.android.trace.impl.internal.DatadogSpanBuilderAdapter
+import com.datadog.android.trace.impl.internal.DatadogSpanContextAdapter
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.trace.bootstrap.instrumentation.api.AgentSpan
+import com.datadog.trace.bootstrap.instrumentation.api.AgentSpanLink
+import com.datadog.trace.bootstrap.instrumentation.api.AgentTracer
+import fr.xgouchet.elmyr.annotation.DoubleForgery
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.annotation.LongForgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+class DatadogSpanBuilderAdapterTest {
+
+    @StringForgery
+    lateinit var fakeString: String
+
+    @LongForgery
+    private var fakeLong: Long = 0
+
+    @DoubleForgery
+    private var fakeDouble: Double = 0.0
+
+    @Forgery
+    lateinit var fakeDatadogSpanLink: DatadogSpanLink
+
+    @Mock
+    private lateinit var mockAgentSpanBuilderAdapter: AgentTracer.SpanBuilder
+
+    @Mock
+    private lateinit var mockAgentSpanContext: AgentSpan.Context
+
+    private lateinit var testedBuilderAdapter: DatadogSpanBuilderAdapter
+
+    @BeforeEach
+    fun `set up`() {
+        testedBuilderAdapter = DatadogSpanBuilderAdapter(mockAgentSpanBuilderAdapter)
+    }
+
+    @Test
+    fun `M delegate ignoreActiveSpan W ignoreActiveSpan is called`() {
+        testedBuilderAdapter.ignoreActiveSpan()
+
+        verify(mockAgentSpanBuilderAdapter).ignoreActiveSpan()
+    }
+
+    @Test
+    fun `M return DatadogSpan W start() is called`() {
+        whenever(mockAgentSpanBuilderAdapter.start()).thenReturn(mock<AgentSpan>())
+
+        val span = testedBuilderAdapter.start()
+
+        assertThat(span).isInstanceOf(DatadogSpan::class.java)
+    }
+
+    @Test
+    fun `M delegate withOrigin W withOrigin is called`() {
+        assertThat(testedBuilderAdapter.withOrigin(fakeString)).isEqualTo(testedBuilderAdapter)
```

Comment:
minor: It is better to extract result of `testedBuilderAdapter.withOrigin(fakeString)` into a variable of `When` block. Same for other similar sites.